src/test/java/seedu/fma/model/exercise/UniqueExerciseListTest.java

```diff
@@ -0,0 +1,170 @@
+        assertThrows(NullPointerException.class, () -> uniqueExerciseList.setExercise(EXERCISE_A, null));
+    }
+
+    @Test
+    public void setExercise_targetExerciseNotInList_throwsExerciseNotFoundException() {
+        assertThrows(ExerciseNotFoundException.class, () -> uniqueExerciseList.setExercise(EXERCISE_A, EXERCISE_A));
+    }
+
+    @Test
+    public void setExercise_editedExerciseIsSameExercise_success() {
+        uniqueExerciseList.add(EXERCISE_A);
+        uniqueExerciseList.setExercise(EXERCISE_A, EXERCISE_A);
+        UniqueExerciseList expectedUniqueExerciseList = new UniqueExerciseList();
+        expectedUniqueExerciseList.add(EXERCISE_A);
+        assertEquals(expectedUniqueExerciseList, uniqueExerciseList);
+    }
+
+    @Test
+    public void setExercise_editedExerciseHasSameIdentity_success() {
+        uniqueExerciseList.add(EXERCISE_A);
+        Exercise editedExerciseA = new ExerciseBuilder(EXERCISE_A)
+                .build();
+        uniqueExerciseList.setExercise(EXERCISE_A, editedExerciseA);
+        UniqueExerciseList expectedUniqueExerciseList = new UniqueExerciseList();
+        expectedUniqueExerciseList.add(editedExerciseA);
+        assertEquals(expectedUniqueExerciseList, uniqueExerciseList);
+    }
+
+    @Test
+    public void setExercise_editedExerciseHasDifferentIdentity_success() {
+        uniqueExerciseList.add(EXERCISE_A);
+        uniqueExerciseList.setExercise(EXERCISE_A, EXERCISE_B);
+        UniqueExerciseList expectedUniqueExerciseList = new UniqueExerciseList();
+        expectedUniqueExerciseList.add(EXERCISE_B);
+        assertEquals(expectedUniqueExerciseList, uniqueExerciseList);
+    }
+
+    @Test
+    public void setExercise_editedExerciseHasNonUniqueIdentity_throwsDuplicateExerciseException() {
+        uniqueExerciseList.add(EXERCISE_A);
+        uniqueExerciseList.add(EXERCISE_B);
+        assertThrows(DuplicateExerciseException.class, () -> uniqueExerciseList.setExercise(EXERCISE_A, EXERCISE_B));
+    }
+
+    @Test
+    public void remove_nullExercise_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueExerciseList.remove(null));
+    }
+
+    @Test
+    public void remove_exerciseDoesNotExist_throwsExerciseNotFoundException() {
+        assertThrows(ExerciseNotFoundException.class, () -> uniqueExerciseList.remove(EXERCISE_A));
+    }
+
+    @Test
+    public void remove_existingExercise_removesExercise() {
+        uniqueExerciseList.add(EXERCISE_A);
+        uniqueExerciseList.remove(EXERCISE_A);
+        UniqueExerciseList expectedUniqueExerciseList = new UniqueExerciseList();
+        assertEquals(expectedUniqueExerciseList, uniqueExerciseList);
+    }
+
+
+    @Test
+    public void setExercises_nullList_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueExerciseList.setExercises((List<Exercise>) null));
+    }
+
+    @Test
+    public void setExercises_list_replacesOwnListWithProvidedList() {
```

Comment:
could rename to listWithSingleExercise (current naming feels like it's a superset of the next test case)